tests/unit/swagger/schemaDetails3.spec.ts

```diff
@@ -18,13 +18,16 @@ describe('Definition generation for OpenAPI 3.0.0', () => {
   const optionsWithNoAdditional = Object.assign<{}, SwaggerConfig, Partial<SwaggerConfig>>({}, defaultOptions, {
     noImplicitAdditionalProperties: 'silently-remove-extras',
   });
+  const optionsWithHostIsNull = Object.assign<{}, SwaggerConfig, Partial<SwaggerConfig>>({}, defaultOptions, {
+    host: null,
+  });
 
   interface SpecAndName {
     spec: Swagger.Spec3;
     /**
      * If you want to add another spec here go for it. The reason why we use a string literal is so that tests below won't have "magic string" errors when expected test results differ based on the name of the spec you're testing.
      */
-    specName: 'specDefault' | 'specWithNoImplicitExtras';
+    specName: 'specDefault' | 'specWithNoImplicitExtras' | 'specWithHostIsNull';
```

Comment:
```suggestion
    specName: 'specDefault' | 'specWithNoImplicitExtras';
```
**Short version:** please initialize `optionsWithHostIsNull` inside of your `'setting host to null should result in relative URL'` test. Do not add it to the `SpecAndName` interface since that's designed to for the purpose of iterating over schema properties.

**Longer version:** Is it necessary to add a new spec type here? The reason why I ask is that this causes another iteration over the `assertionsPerProperty` record (if it were added to [that array](https://github.com/lukeautry/tsoa/pull/547/files#diff-19811e20ee8ebdc5405ac00d6cd74d94L56))... which is a lot of tests. i.e. it takes a long time. I'm talking about this part:
```ts
const assertionsPerProperty: Record<keyof TestModel, (propertyName: string, schema: Swagger.Spec) => void> 
```

And in the case of your change, there's no reason why the host would change the property schemas. So it doesn't need to be tested in that manner. So `optionsWithHostIsNull` doesn't need to live in the outer scope. It only is used within the `'setting host to null should result in relative URL'` test and that's fine. Let's keep it private to there.

Reply:
ok, i will adapt the PR.
i kind of missed the usecase of specName, i thought this was to collect all variants for different schemas and to have them always checked, just to be sure.